ui/tooluifactory.cpp

```diff
@@ -50,4 +73,37 @@ bool ToolUiFactory::remotingSupported() const
 
 void ToolUiFactory::initUi()
 {
+
+}
+
+void ToolUiFactory::initUiObject()
+{
+    if (s_ObjectFactoriesInitialized)
+        return;
+    s_ObjectFactoriesInitialized = true;
+
+    PropertyWidget::registerTab<PropertiesTab>(QStringLiteral("properties"),
+                                               ObjectInspectorFactory::tr("Properties"),
+                                               PropertyWidgetTabPriority::First);
+    ObjectBroker::registerClientObjectFactoryCallback<PropertiesExtensionInterface *>(
+        createExtension<PropertiesExtensionClient>);
+    PropertyWidget::registerTab<MethodsTab>(QStringLiteral("methods"),
+                                            ObjectInspectorFactory::tr("Methods"),
+                                            PropertyWidgetTabPriority::Basic - 1);
+    ObjectBroker::registerClientObjectFactoryCallback<MethodsExtensionInterface *>(
+        createExtension<MethodsExtensionClient>);
+    PropertyWidget::registerTab<ConnectionsTab>(QStringLiteral("connections"),
+                                                ObjectInspectorFactory::tr("Connections"),
+                                                PropertyWidgetTabPriority::Basic - 1);
+    ObjectBroker::registerClientObjectFactoryCallback<ConnectionsExtensionInterface *>(
+        createExtension<ConnectionsExtensionClient>);
+    PropertyWidget::registerTab<EnumsTab>(QStringLiteral("enums"),
+                                          ObjectInspectorFactory::tr("Enums"),
+                                          PropertyWidgetTabPriority::Exotic - 1);
+    PropertyWidget::registerTab<ClassInfoTab>(QStringLiteral("classInfo"),
+                                              ObjectInspectorFactory::tr("Class Info"),
+                                              PropertyWidgetTabPriority::Exotic - 1);
+    PropertyWidget::registerTab<ApplicationAttributeTab>(
+                QStringLiteral("applicationAttributes"),
+                ObjectInspectorFactory::tr("Attributes"), PropertyWidgetTabPriority::Advanced);
```

Comment:
Does that mean that initUi() isn't called for the object inspector as soon as the object tool is activated (not selected by the user, but becomes selectable as the corresponding part on the probe sees the first matching object)? That would be the actual bug then, initUi is separate from the widget ctor exactly for this reason, so we can init property tabs even if the tool isn't actually visible.